core/lib/slider.ts

```diff
@@ -0,0 +1,569 @@
+import {computed, writable} from '@amadeus-it-group/tansu';
+import type {WidgetsCommonPropsAndState} from './commonProps';
+import type {ConfigValidator, PropsConfig} from './services';
+import {bindableDerived, registrationArray, writablesForProps} from './services';
+import type {ReadableSignals} from './services/stores';
+import {mergeConfigStores, normalizeConfigStores, stateStores} from './services/stores';
+import {typeArray, typeBoolean, typeFunction, typeNumber, typeString} from './services/writables';
+import type {Directive, Widget} from './types';
+import {noop} from './utils';
+
+export interface SliderCommonPropsAndState extends WidgetsCommonPropsAndState {
+	/**
+	 * Minimum value that can be assigned to the slider
+	 */
+	min: number;
+
+	/**
+	 * Maximum value that can be assigned to the slider
+	 */
+	max: number;
+
+	/**
+	 * Unit value between slider steps
+	 */
+	stepSize: number;
+
+	/**
+	 * If `true` slider value cannot be changed but the slider is still focusable
+	 */
+	readonly: boolean;
+
+	/**
+	 * If `true` slider value cannot be changed and the slider cannot be focused
+	 */
+	disabled: boolean;
+
+	/**
+	 * If `true` is vertically positioned otherwise it is horizontal
+	 */
+	vertical: boolean;
+
+	/**
+	 * Dom Rectangle of the slider
+	 */
+	sliderDomRect: DOMRect;
+
+	/**
+	 * Dom Rectangle of the minimum label
+	 */
+	minLabelDomRect: DOMRect;
+
+	/**
+	 * Dom Rectangle of the maximum label
+	 */
+	maxLabelDomRect: DOMRect;
+
+	/**
+	 * Current slider values
+	 */
+	values: number[];
+}
+
+export interface SliderState extends SliderCommonPropsAndState {
+	/**
+	 * Progress bar left position in %
+	 */
+	progressLeft: number[];
+
+	/**
+	 * Progress bar bottom position in %
+	 */
+	progressBottom: number[];
+
+	/**
+	 * Progress bar height in %
+	 */
+	progressHeight: number[];
+
+	/**
+	 * Progress bar width in %
+	 */
+	progressWidth: number[];
+
+	/**
+	 * Sorted slider values
+	 */
+	sortedValues: number[];
+
+	/**
+	 * Mix label left offset in %
+	 */
+	mixLabelLeft: number;
+
+	/**
+	 * Mix label top offset in %
+	 */
+	mixLabelTop: number;
+
+	/**
+	 * Array of the left offsets for the tooltips in %
+	 */
+	handleTooltipLeft: number[];
+
+	/**
+	 * Array of the top offsets for the tooltips in %
+	 */
+	handleTooltipTop: number[];
+
+	/**
+	 * If `visible` the minimum label will be displayed
+	 */
+	minValueLabelDisplay: string;
```

Comment:
Also maybe `minValueLabelVisible` or `minValueLabelDisplayed` or similar in this case?

Reply:
Also why changing visibility and not removing it from DOM ? (I think I left similar comment below.

Reply:
I am not removing it from the dom as the resizeobserver looks for the element in the DOM, so if we remove it will be undefined and will break the computations
The strings are used because the style accepts 'hidden' or 'visible' and not the true/false, so to avoid conversion I chose strings

Reply:
You already have a conversion from boolean to string in the core, just move it to the template if you really need to use the style.
But if you move the ResizeObserver to the core as I am advising you, you will have to handle the case of missing DOM elements anyway and you could as well remove the elements from the DOM instead of hiding them.